src/notebook/cells/widget.ts

```diff
@@ -585,20 +559,14 @@ namespace CodeCellWidget {
    * The default implementation of an `IRenderer`.
    */
   export
-  class Renderer extends BaseCellWidget.Renderer implements IRenderer {
+  abstract class Renderer extends BaseCellWidget.Renderer implements IRenderer {
```

Comment:
In this case for my editor i have to duplicate code to instantiate input and output areas, they suit me well, how does it harm?